src/NSwag.Generation.WebApi/WebApiOpenApiDocumentGenerator.cs

```diff
@@ -290,18 +290,32 @@ private static IEnumerable<MethodInfo> GetActionMethods(Type controllerType)
                 });
         }
 
-        private string GetOperationId(OpenApiDocument document, string controllerName, MethodInfo method)
+        private string GetOperationId(OpenApiDocument document, string controllerName, MethodInfo method, string httpMethod)
         {
             string operationId;
 
             dynamic swaggerOperationAttribute = method
                 .GetCustomAttributes()
                 .FirstAssignableToTypeNameOrDefault("SwaggerOperationAttribute", TypeNameStyle.Name);
 
+            dynamic httpAttribute = null;
+            if (!string.IsNullOrWhiteSpace(httpMethod))
+            {
+                var attributeName = Char.ToUpperInvariant(httpMethod[0]) + httpMethod.Substring(1).ToLowerInvariant();
+                var typeName = string.Format("Microsoft.AspNetCore.Mvc.Http{0}Attribute", attributeName);
```

Comment:
Sorry, but this is the legacy Web API generator which should not be used for ASP.NET Core... I think you're using the wrong generator in your project :-)

Reply:
I was doing it because the code was equal to ASP.NET Core, but just to have it for both running.
The method from which i am retrieving the name is already present in the code.

see https://docs.microsoft.com/en-us/aspnet/web-api/overview/web-api-routing-and-actions/attribute-routing-in-web-api-2#adding-route-attributes